android/library/src/main/java/com/polidea/multiplatformbleadapter/utils/OneTimeActionExecutor.java

```diff
@@ -0,0 +1,15 @@
+package com.polidea.multiplatformbleadapter.utils;
+
+public abstract class OneTimeActionExecutor<T> {
```

Comment:
Yes, it solves the problem of calling onErrorCallback too many times. In some places, we emit cancellation error on unsubscribe event. Since doOnUnsubscribe is called right after doOnError, we are calling onErrorCallback with actual error and then with cancellation error. 
After discussion with @dariuszseweryn we decided that it is an unwanted behavior. Moreover calling onErrorCallback more than once forces us to make workarounds as SafePromises on react-native side. Having this solution implemented in adapter, we can get rid of SafePromises in RN lib.

It is used couple of times in BleModule, for example inside connectToDevice or enable methods.

Reply:
@Cierpliwy talked about a few scenarios, where callbacks might be called twice, while they should be called once. Those were:
1. Success-success
2. Success-error
3. Error-success
4. Error-error

This solves only the fourth issue.

Reply:
Ok, so in RN we still need to handle this by using SafePrimises mechanism, but for the sake of keeping proper errors emitting policy we should also leave this OneTimeActionExecutor in the adapter.